### PR TITLE
Optimize Rtypes.h include

### PR DESCRIPTION
**Pull request by linev**
There are many standard include in Rtypes.h like `stdio.h` or `string.h` or `snprintf.h`
All these includes appears in 99% of ROOT sources - without real use
Try to minimize usage of such includes in headers, moving them to source where needed
Also replace `Rtypes.h` by `RtypesCore.h` where it make sense.
Introduce `strtok.h` include which provides `R__STRTOK_R` method, use it only where really required

### CODE REVIEW

File: core/base/inc/Rtypes.h [hunk 1]
```diff
@@ -320,27 +322,27 @@ public: \
    virtual_keyword void Streamer(TBuffer &R__b) overrd { ::ROOT::Internal::DefaultStreamer(R__b, name::Class(), this); }
 
 #define ClassDef(name,id) \
-   _ClassDefOutline_(name,id,virtual,)   \
+   _ClassDefOutline_(name,id,virtual,) \
    static int DeclFileLine() { return __LINE__; }
```
Comment (pcanal) [inline core/base/inc/Rtypes.h:326]:
```suggestion
 #define ClassDef(name,id)                             \
    _ClassDefOutline_(name,id,virtual,)             \
   static int DeclFileLine() { return __LINE__; }
```
is likely more consistent with the intent/rest-of-code (and in my opinion somewhat easier to read).

Comment (linev) [reply core/base/inc/Rtypes.h]: @pcanal I rebase code and adjust `Rtypes.h` format

**Comment by linev** [comment]
@pcanal Now PR is ready for review.

**Comment by vgvassilev** [comment]
Did you make sure it works for -Dcxxmodules=On?

**Comment by linev** [comment]
> Did you make sure it works for -Dcxxmodules=On?

How I can check it? I cannot compile current master when specifying `-Dcxxmodules=On`.
Same will be after I merge this PR.

There is simple error which can be fixed in graf2s/mathtext/inc/mathrenderer.h, but it does not solve general compilation errors with cxxmodules. I test both on mac and linux platforms

**Comment by vgvassilev** [comment]
Fair enough. As long as it does not make it worse I am happy.

**Comment by linev** [comment]
I submit PR #5812 fixing compilation error with `cxxmodules` in mathtext. 
But main problem is clang - many errors appears there.

**Comment by pcanal** [comment]
> How I can check it? I cannot compile current master when specifying -Dcxxmodules=On.

This is a disaster as module as now the default.   Is it failing for you when building from scratch?  Which plaftorm and compiler?

**Comment by vgvassilev** [comment]
That’s not the runtime_cxxmodules but just cxxmodules. Different feature ;)

**Pull request merged**